apidocs: Display response codes in example responses.

MSurfer20 opened the pull request:
Currently, the examples response don't show response
codes anywhere, which is already existing and important
information. This commit adds the same.
![Screenshot from 2021-07-08 21-21-25](https://user-images.githubusercontent.com/56172924/124953570-e7fc7280-e032-11eb-9f53-54e8155d70e1.png)

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing plan:** <!-- How have you tested? -->


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

MSurfer20 commented:
@timabbott  We could also use the colour we use for Parameter data types here, although I feel that the above one might look better. The alternate colour looks like:
![Screenshot from 2021-07-08 21-19-42](https://user-images.githubusercontent.com/56172924/124953757-0a8e8b80-e033-11eb-82c8-dabecbb56049.png)

andersk commented on zerver/openapi/openapi.py:
```diff
@@ -256,9 +256,9 @@ def get_parameters_description(endpoint: str, method: str) -> str:
 def generate_openapi_fixture(endpoint: str, method: str, status_code: str = "200") -> List[str]:
     """Generate fixture to be rendered"""
-    fixture = []
+    fixture = [f"#### {status_code} Response(s)"]
```
Response [should not](https://zulip.readthedocs.io/en/latest/translating/translating.html#capitalization) be capitalized here.

MSurfer20 committed:
apidocs: Display title for each response fixture with HTTP code.

Currently, we didn't provide a heading and response
code information along with example responses. whereas
a suitable title can be generated from code parameter in
responses and the HTTP codes already exist in OpenAPI spec.

This commit adds support to display the same. This also removes
already present headings from some response descriptions.

MSurfer20 commented:
New design:
![image](https://user-images.githubusercontent.com/56172924/125089543-125e3680-e0ec-11eb-9515-42b06df373cb.png)

timabbott commented:
I think we need a bit more margin between the heading and the status code.

I also am not sure about the removal of the heading text for the common errors content; can you post a screenshot?

MSurfer20 commented:
@timabbott Some of them look good, while in some cases, the information did end up being lost:
![image](https://user-images.githubusercontent.com/56172924/125571606-eea0f061-e3e9-476c-a9cb-6755376c8d5e.png)
![image](https://user-images.githubusercontent.com/56172924/125571626-e691b006-cafc-4e7e-b112-332cd6716c1c.png)

MSurfer20 commented:
@timabbott A ping regarding this in case you lost track of it.
